dotCMS/src/main/java/com/dotmarketing/portlets/osgi/AJAX/OSGIAJAX.java

```diff
@@ -48,8 +48,25 @@ public void undeploy ( HttpServletRequest request, HttpServletResponse response
 
         //Then move the bundle from the load folder to the undeployed folder
 
-        File from = new File( FileUtil.getRealPath( "/WEB-INF/felix/load/" + jar ) );
-        File to = new File( FileUtil.getRealPath( "/WEB-INF/felix/undeployed/" + jar ) );
+        String loadPath;
+        String undeployedPath;
+
+        try {
+            loadPath = OSGIUtil.getInstance().getBundleContext().getProperty(OSGIUtil.FELIX_FILEINSTALL_DIR);
+        } catch (Exception ex) {
+            Logger.error(this, "Unable to find the felix load folder path from OSGI bundle context. Setting it manually to /WEB-INF/felix/load", ex);
+            loadPath = "/WEB-INF/felix/load";
+        }
+
+        try {
+            undeployedPath = OSGIUtil.getInstance().getBundleContext().getProperty(OSGIUtil.FELIX_UNDEPLOYED_DIR);
+        } catch (Exception ex) {
+            Logger.error(this, "Unable to find the felix undeployed folder path from OSGI bundle context. Setting it manually to /WEB-INF/felix/undeployed", ex);
+            undeployedPath = "/WEB-INF/felix/undeployed";
+        }
```

Comment:
Same as above with this one. Extract the full try-catch block to a private method "getUndeployedPath" so both deploy and undeploy can use it.